text/0000-lwc-signals.md

```diff
@@ -0,0 +1,373 @@
+
+## Drawbacks
+
+### Increased Complexity for Developers
+
+* __Explicit Reactivity Management__: The shift from `@track`'s implicit reactivity to explicit signals and stores introduces additional complexity. For thos developers that are still using `@track`, they must now manage subscriptions and updates more directly, which could increase the cognitive load, compared to the simplicity of `@track`.
+* __Learning Curve__: There's an inherent learning curve as developers familiarize themselves with the new patterns of Signals and stores, particularly around the nuances of subscription management and the .value convention.
+
+### Migration Effort
+
+* __Codebase Refactoring__: Transitioning existing code that relies on `@track` and `@wire` to the new model will require careful refactoring. This could be a significant effort for large and complex applications.
+* __Backward Compatibility__: Ensuring backward compatibility during the transition period may necessitate maintaining dual systems, complicating the framework's implementation and usage.
+* __Verbose Code__: Some scenarios, like converting `@wire` methods to store subscriptions, result in more verbose code, which might detract from the developer experience.
+
+### Tooling and Documentation Updates
+
+* __Development Tooling__: All tooling, including IDEs, linters, and debuggers, will need updates to accommodate and optimize for Signals and stores.
+* __Educational Materials__: Comprehensive updates to documentation, tutorials, and example code will be essential to support developers in adopting the new model.
+
+### Potential for Misuse
+
+* __Overuse of Signals__: There's a risk that developers might overuse signals for state management, leading to unnecessarily reactive code, which can be less performant and harder to maintain. This is specially relevant because of the runtime detection of `.value` to maintain backward compatibility with existing components. Any misused on the signals can affect the runtime performance of the component and therefore the overall performance of the application.
+* __Subscription Management__: Incorrect handling of subscriptions could introduce memory leaks or lead to unexpected behaviors if not managed correctly. This is specially important because the store implementation doesn't have access anymore to the component's lifecycle hooks, instead it would just stop receiving updates of the stores piped thru `this.$api`, meaning that we are at the mercy of the garbage collector.
```

Comment:
This is the one that concerns me the most. I can see *lots* of component authors messing up subscriptions. `@wire` has issues, but one of the nice things about it is that there's no way for a component author to mess up the subscription.